lnwallet/interface.go

```diff
@@ -592,6 +595,50 @@ type MessageSigner interface {
 		doubleHash bool) (*ecdsa.Signature, error)
 }
 
+// AddrWithKey wraps a normal addr, but also includes the internal key for the
+// delivery addr if known.
+type AddrWithKey struct {
+	lnwire.DeliveryAddress
+
+	InternalKey fn.Option[btcec.PublicKey]
+
+	// TODO(roasbeef): consolidate w/ instance in chan closer
+}
+
+// InternalKeyForAddr returns the internal key associated with a taproot
+// address.
+func InternalKeyForAddr(wallet WalletController, netParams *chaincfg.Params,
```

Comment:
The version you replaced in the peer package had a bunch of bug fixes in it that allowed the unit tests to pass.

Reply:
Also, we have cases where we actually don't know (and don't care about) the internal key, for example when it's not a p2tr key). So returning `None` here is a valid use case, so `Result` IMO doesn't really make sense. Or, in fact, being able to return `None` is needed, since this code will also run for non-custom channels and then fail due to the "expected taproot addr".

Reply:
> The version you replaced in the peer package had a bunch of bug fixes in it that allowed the unit tests to pass.

Ah, will circle back on that, just now getting the PR to function properly outside of the itests elsewhere.

Reply:
> Or, in fact, being able to return None is needed, since this code will also run for non-custom channels and then fail due to the "expected taproot addr".

Correct that there's a gap here: if a user passes in a none taproot addr, or really just an exterrnal addr, then we'll fail here as is. For addrs from the wallet, they should always be P2TR addresses however. 

Re `None` vs otherwise, if we relax this to return a nil error, then callers can get a `None` option from the return value. Still feeling out what "feels" good with the `Result` API, but agree that can be a bit awkward. Will take a look at restoring those bug fixes and also going back to the pure option return value. 

I think I shifted things as I started messing with the chained `Result` pattern, where you can just pipeline the computation and only have to worry about the error at the very end with a call to `Unpack()`.

Reply:
FWIW, went back w/ just the option variant here in the latest diff.